mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -139,15 +191,47 @@ class OrderedTrackedLayout[T: ClassTag](values: Traversable[(ReferenceRegion, T)
   class Track(val initial: (ReferenceRegion, T)) {
 
     val records = new mutable.ListBuffer[(ReferenceRegion, T)]()
-    records += initial
+    val groupPairs = new mutable.ListBuffer[GroupPair]()
+    val idx: Int = numTracks
+
+    if (initial != null)
+      records += initial
+
+    def this() {
+      this(null)
+    }
 
     def +=(rec: (ReferenceRegion, T)): Track = {
       records += rec
       this
     }
 
+    def addGroupPair(recs: Seq[(ReferenceRegion, T)]) {
+      if (recs.size < 2)
+        return
+      else if (recs.size == 2)
+        createGroupPair(recs(0), recs(1))
+      else
+        log.info("Warning: Group Pairs do not support > 2 pairs per read")
```

Comment:
@erictu what are your thoughts about warning here vs throwing an exception?

BTW, you can do `log.warn` here instead and then remove the `Warning:` from the message. The different logger methods (e.g., `info`, `warn`, `error`, `trace`) map to different logging levels in [`SLF4j`](http://www.slf4j.org/) and allow the user to decide what level of messages they want to see.